experimental/examples/imageviewer/shared/src/commonMain/kotlin/example/imageviewer/Dependencies.kt

```diff
@@ -1,80 +1,102 @@
 package example.imageviewer
 
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import androidx.compose.runtime.staticCompositionLocalOf
 import androidx.compose.ui.graphics.ImageBitmap
-import example.imageviewer.core.BitmapFilter
-import example.imageviewer.core.FilterType
-import example.imageviewer.model.ContentRepository
-import example.imageviewer.model.Picture
-import example.imageviewer.model.name
-import io.ktor.client.*
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.serialization.json.Json
-
-interface Dependencies {
-    val httpClient: HttpClient
-    val ioScope: CoroutineScope
-    fun getFilter(type: FilterType): BitmapFilter
-    val localization: Localization
-    val imageRepository: ContentRepository<ImageBitmap>
-    val notification: Notification
-    val json: Json get() = jsonReader
+import example.imageviewer.model.*
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emptyFlow
+import org.jetbrains.compose.resources.ExperimentalResourceApi
+import org.jetbrains.compose.resources.resource
+
+@OptIn(ExperimentalResourceApi::class)
+abstract class Dependencies {
+    abstract val notification: Notification
+    abstract val imageStorage: ImageStorage
+    val pictures: SnapshotStateList<PictureData> = mutableStateListOf(*resourcePictures)
+    open val externalEvents: Flow<ExternalImageViewerEvent> = emptyFlow()
+    val localization: Localization = when (getCurrentLanguage()) {
+        AvailableLanguages.EN -> EnglishLocalization
```

Comment:
Let's add converter in `Localization.kt` instead

Reply:
Done